### PR TITLE
SPCTR-804 : Added Responsive Stacking, Bug fixes

### PR DESCRIPTION
**Pull request by aapatre**
### Description
- Added responsive stacking feature (upgraded previous stacking multi-button to responsive)
- Fixed vertical alignment of stars and title
- Made changes and new additions to alignment code since `display: flex` is now used for some elements and there were breaking changes that needed to be fixed

### Issue References
- [JIRA](https://brainstormforce.atlassian.net/browse/SPCTR-804)
- [R3 Docs](https://docs.google.com/document/d/1jGAQGdVevCytKOb6pihU1ZPURFFKs0PanrOG97Xxj-o/edit#heading=h.eo8f2i54l70p)

### Types of changes
- Bug fix (non-breaking change which fixes an issue)
- New feature (upgraded Stacking to Responsive Stacking)

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Review by bsf-zanev — APPROVED** [review]
All Updated Code Reviewed and Approved.

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: includes/blocks/star-rating/frontend.css.php [hunk 1]
```diff
@@ -29,6 +29,30 @@
 	}
 }
 
+if ( ! function_exists( 'flex_alignment' ) ) {
+
+	/** Since title text is set to flex, we need this function so that stack alignment doesn't break.
+	 * It converts the normal text-align values to flex-alignment based values.
+	 *
```
[the remark this answers is not in the document]
Comment (aapatre) [reply includes/blocks/star-rating/frontend.css.php]: Done

File: classes/class-uagb-block-helper.php [hunk 2]
```diff
@@ -1072,5 +1072,27 @@ public static function uag_get_background_obj( $bg_obj ) {
 
 			return $gen_bg_css;
 		}
+
+		/**
+		 * Since title text is set to flex, we need this function so that stack alignment doesn't break.
+		 * It converts the normal text-align values to flex-alignment based values.
+		 *
```
Comment (RajkiranBagal) [inline classes/class-uagb-block-helper.php:1079]: add a since tag.

File: classes/class-uagb-block-helper.php [hunk 3]
```diff
@@ -1072,5 +1072,27 @@ public static function uag_get_background_obj( $bg_obj ) {
...
+		 * @param string $textAlign Alignment value from text-align property.
+		 */
+		public static function flex_alignment( $textAlign ) {
+
+			switch ( $textAlign ) {
```
Comment (RajkiranBagal) [inline classes/class-uagb-block-helper.php:1084]: php variables should always be `snake_case`

File: classes/class-uagb-block-helper.php [hunk 4]
```diff
@@ -1072,5 +1072,27 @@ public static function uag_get_background_obj( $bg_obj ) {
...
 			return $gen_bg_css;
 		}
+
+		/**
+		 * Since title text is set to flex, we need this function so that stack alignment doesn't break.
+		 * It converts the normal text-align values to flex-alignment based values.
+		 *
+		 * @param string $textAlign Alignment value from text-align property.
+		 */
+		public static function flex_alignment( $textAlign ) {
```
Comment (RajkiranBagal) [inline classes/class-uagb-block-helper.php:1082]: add meaningful function name so anyone can understand what it does.

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Comment by aapatre** [comment]
Ensured that all the comments are resolved and all the requested changes are made :+1:

**Review by RajkiranBagal — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**